packages/@ember/-internals/glimmer/lib/component-managers/curly.ts

```diff
@@ -92,7 +93,7 @@ function applyAttributeBindings(
     operations.setAttribute('id', PrimitiveReference.create(id), false, null);
   }
 
-  if (seen.indexOf('style') === -1) {
+  if (EMBER_COMPONENT_IS_VISIBLE && IsVisibleBinding && seen.indexOf('style') === -1) {
```

Comment:
Let's try to avoid the boolean coercion here:

```suggestion
  if (EMBER_COMPONENT_IS_VISIBLE && IsVisibleBinding !== undefined && seen.indexOf('style') === -1) {
```